src/Resources/views/CRUD/list_outer_rows_mosaic.html.twig

```diff
@@ -62,16 +66,15 @@ This template can be customized to match your needs. You should only extends the
                             {% endif %}
 
                             {% block sonata_mosaic_description %}
-                                {% if admin.hasAccess('edit', object) and admin.hasRoute('edit') %}
-                                    <a class="mosaic-inner-link" href="{{ admin.generateUrl('edit', {'id' : object|sonata_urlsafeid(admin) }) }}">{{ meta.title|truncate(40) }}</a>
-                                {% elseif admin.hasAccess('show', object) and admin.hasRoute('show') %}
-                                    <a class="mosaic-inner-link" href="{{ admin.generateUrl('show', {'id' : object|sonata_urlsafeid(admin) }) }}">{{ meta.title|truncate(40) }}</a>
-                                {% else %}
-                                    {{ meta.title|truncate(40) }}
-                                {% endif %}
+                                {{ meta.title|truncate(40) }}
                             {% endblock %}
                         </div>
                     </div>
+                    {% if (admin.hasAccess('edit', object) and admin.hasRoute('edit'))
+                        or (admin.hasAccess('show', object) and admin.hasRoute('show'))
+                    %}
+                        </a>
+                    {% endif %}
```

Comment:
This looks dangerous, please use one condition and a block variable:
```html+jinja
{% set foo %}
some html
{% endset%}
{% if some condition %}
<a>{{foo}}</a>
{% else %}
{{foo}}
{% endif %}


```

Reply:
Agree, looks cleaner, will be fixed!